UPGRADING.rst

```diff
@@ -11,6 +11,20 @@ reverted. This is all fairly informal and loosely defined. Hopefully we won't
 have too many entries in this file.
 
 
+#4691 Fix: S3 Block Public Access setting should be enabled at the bucket-level
```

Comment:
I think this should be clearer about which upgrade steps are specific to the operator and which aren't. 

Presumably, only the operator will apply the changes to the shared components. But the last sentence implies that everyone else needs to perform some sort of upgrade step as well, without explaining what that step is.

As a non-operator, am I supposed to run `make -C $project_root/terraform/shared apply` for my personal deployments? Do I just run `make deploy`? 

My understanding is that UPGRADING.txt is written to address all developers, not just the operator. I think everyone should be able to tell what steps they need to follow from reading the document, without being asked.

Reply:
Yeah, I assumed that mentioning updates to the main Terraform component conveyed the need for running the make deploy target, when it was actually ambiguous. Previous feedback suggested that adding upgrading instructions to run`make deploy` targets was redundant, which is why I was hesitant to include it. I think is clearer now, lmkwyt.

Reply:
I agree that having instructions to run `make deploy` is redundant but `make clean` is novel to me so I think this is good.

Reply:
Mentioning `make deploy` is generally redundant , except here we want to make sure that people update their deployments ASAP.